util/src/test/java/io/camunda/zeebe/util/sched/future/TestActorFuture.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.util.sched.future;
+
+import io.camunda.zeebe.util.Either;
+import io.camunda.zeebe.util.sched.ActorTask;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.BiConsumer;
+import org.agrona.LangUtil;
+
+/**
+ * Implementation of {@code ActorFuture} for use in tests. The main goal is to use this in tests
+ * without starting the actor scheduler.
+ *
+ * <p>The implementation is limited - only the methods currently needed in tests are implemented,
```

Comment:
:wrench: Could throw `UnsupportedOperationException` instead of `IllegalStateException`, which I think is more common for such things.